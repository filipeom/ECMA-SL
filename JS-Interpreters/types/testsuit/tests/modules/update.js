a = 3;  // retval = 3
b = 2;  // retval = 2
++a;    // retval = 4
--b;    // retval = 1
a++;    // retval = 4
b--;    // retval = 1
a;      // retval = 5
b;      // retval = 0



// retval = 0