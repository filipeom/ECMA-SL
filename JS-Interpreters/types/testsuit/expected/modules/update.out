PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Program [stmts]:"
PROGRAM PRINT: ["$loc_3", "$loc_7", "$loc_11", "$loc_14", "$loc_17", "$loc_20", "$loc_23", "$loc_25"]
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_4" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "type": "AssignmentExpression", "right": "$loc_6", "left": "$loc_5", "operator": "=" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "AssignmentExpression [left][right]:"
PROGRAM PRINT: { "name": "a", "type": "Identifier" }
PROGRAM PRINT: { "type": "Literal", "raw": "3", "value": 3 }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 3.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 3.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_8" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "type": "AssignmentExpression", "right": "$loc_10", "left": "$loc_9", "operator": "=" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "AssignmentExpression [left][right]:"
PROGRAM PRINT: { "name": "b", "type": "Identifier" }
PROGRAM PRINT: { "type": "Literal", "raw": "2", "value": 2 }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Literal [value]"
PROGRAM PRINT: 2.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 2.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_12" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "prefix": true, "type": "UpdateExpression", "argument": "$loc_13", "operator": "++" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "UpdateExpressionInc [arg]"
PROGRAM PRINT: { "name": "a", "type": "Identifier" }
PROGRAM PRINT: { "type": "EmptyValue" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 4.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_15" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "prefix": true, "type": "UpdateExpression", "argument": "$loc_16", "operator": "--" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "UpdateExpressionDec [arg]"
PROGRAM PRINT: { "name": "b", "type": "Identifier" }
PROGRAM PRINT: { "type": "EmptyValue" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 1.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_18" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "prefix": false, "type": "UpdateExpression", "argument": "$loc_19", "operator": "++" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "UpdateExpressionInc [arg]"
PROGRAM PRINT: { "name": "a", "type": "Identifier" }
PROGRAM PRINT: { "type": "EmptyValue" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 4.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_21" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "prefix": false, "type": "UpdateExpression", "argument": "$loc_22", "operator": "--" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "UpdateExpressionDec [arg]"
PROGRAM PRINT: { "name": "b", "type": "Identifier" }
PROGRAM PRINT: { "type": "EmptyValue" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 1.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_24" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "name": "a", "type": "Identifier" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Identifier [name][value]:"
PROGRAM PRINT: "a"
PROGRAM PRINT: 5.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 5.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "Start of the statement [stmt]"
PROGRAM PRINT: { "type": "ExpressionStatement", "expression": "$loc_26" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "ExpressionStatement [expression]:"
PROGRAM PRINT: { "name": "b", "type": "Identifier" }
PROGRAM PRINT: "........................................"
PROGRAM PRINT: "Identifier [name][value]:"
PROGRAM PRINT: "b"
PROGRAM PRINT: 0.
PROGRAM PRINT: "End of the statement [retval]"
PROGRAM PRINT: 0.
PROGRAM PRINT: "________________________________________________________________________________"
PROGRAM PRINT: "End of the program [retval]:"
PROGRAM PRINT: 0.
